flutter_bird_app/lib/controller/authentication_service.dart

```diff
@@ -135,54 +153,60 @@ class AuthenticationServiceImpl implements AuthenticationService {
     bool isAuthenticated = recoveredAddress.toLowerCase() == address.toLowerCase();
 
     // Set authenticated account
-    _authenticatedAccount = isAuthenticated ? Account(address: recoveredAddress, chainId: chainId) : null;
+    _authenticatedAccount = isAuthenticated ? Account(address: recoveredAddress, chainId: currentChain!) : null;
 
     return isAuthenticated;
   }
 
   @override
   unauthenticate() async {
-    await _connector?.killSession();
+    if (currentSession != null) {
+      await _connector?.disconnectSession(topic: currentSession!.topic, reason: Errors.getSdkError(Errors.USER_DISCONNECTED));
+    }
     _authenticatedAccount = null;
     _connector = null;
     webQrData = null;
   }
 
   /// Creates a WalletConnect Instance
-  _createConnector({WalletProvider? walletProvider}) async {
+  Future<void> _createConnector({WalletProvider? walletProvider}) async {
     // Create WalletConnect Connector
-    _connector = WalletConnect(
-      bridge: walletConnectBridge,
-      clientMeta: const PeerMeta(
-        name: 'Flutter Bird',
-        description: 'WalletConnect Developer App',
-        url: 'https://flutterbird.com',
-        icons: [
-          'https://raw.githubusercontent.com/Tonnanto/flutter-bird/v1.0/flutter_bird_app/assets/icon.png',
-        ],
-      ),
-    );
+    try {
+      _connector = await Web3App.createInstance(
+        projectId: dotenv.env['WALLET_CONNECT_PROJECT_ID']!,
+        metadata: const PairingMetadata(
+          name: 'Flutter Bird',
+          description: 'WalletConnect Developer App',
+          url: 'https://flutterbird.com',
```

Comment:
Is 'https://flutterbird.com' safe? Could it be that a callback is being sent here to authenticate the wallet?

Reply:
This is not the URL where the callback is sent to authenticate the wallet, but the identifier of the app that connects to the wallet. If this URL is different from the actual URL, the wallet will warn you. Since we already have the URL, let's change it to "https://flutter-bird.netlify.app".